test/run

```diff
@@ -195,60 +217,104 @@ test_from_dockerfile(){
   check_result $?
 }
 
-# Since we built the candidate image locally, we don't want S2I attempt to pull
-# it from Docker hub
-s2i_args="--pull-policy=never"
+test_application_with_user() {
+  # test application with random user
+  CONTAINER_ARGS="--user 12345" test_application
 
-# Verify the 'usage' script is working properly when running the base image with 's2i usage ...'
-test_s2i_usage
-check_result $?
+}
 
-# Verify the 'usage' script is working properly when running the base image with 'docker run ...'
-test_docker_run_usage
-check_result $?
+test_application_enable_init_wrapper() {
+  # test application with init wrapper
+  CONTAINER_ARGS="-e ENABLE_INIT_WRAPPER=true" test_application
+}
+
+test_scl_variables_in_dockerfile() {
+  if [ "$OS" == "rhel7" ] || [ "$OS" == "centos7" ]; then
+     # autocleanup only enabled here as only the following tests so far use it
+     CID_FILE_DIR=$(mktemp -d)
+     ct_enable_cleanup
 
+     info "Testing variable presence during \`docker exec\`"
+     ct_check_exec_env_vars
+     check_result $?
+
+     info "Checking if all scl variables are defined in Dockerfile"
+     ct_check_scl_enable_vars
+     check_result $?
+  fi
+}
+
+function run_all_tests() {
+  local APP_NAME="$1"
+  for test_case in $TEST_SET; do
+    info "Running test $test_case ... "
+    TESTCASE_RESULT=0
+    $test_case
+    local test_msg
+    if [ $TESTCASE_RESULT -eq 0 ]; then
+      test_msg="[PASSED]"
+    else
+      test_msg="[FAILED]"
+      TESTSUITE_RESULT=1
+    fi
+    printf -v test_short_summary "%s %s for '%s' %s\n" "${test_short_summary}" "${test_msg}" "${APP_NAME}" "$test_case"
+    [ -n "${FAIL_QUICKLY:-}" ] && {
+      cleanup "${APP_NAME}"
+      return 1
+    }
+  done;
+}
+
+function test_run_s2i_build() {
+  local test_msg
+  local result="$1"
+  local app="$2"
+  if [[ "$app" == *"-should-fail-"* ]]; then
```

Comment:
If an app fails to build and we expect that, all following tests should be skipped.